tests/vuefire.spec.js

```diff
@@ -25,6 +25,30 @@ describe('VueFire', function () {
     })
   })
 
+  describe('support Function options', function () {
+    it('returns correct ref on function call', function (done) {
+      var firebaseDb = firebaseApp.database()
+      new Vue({
+        data: function () {
+          return {
+            database: firebaseDb
+          }
+        },
+        components: {
+          'child-component': Vue.extend({
+            name: 'ChildComponent',
+            firebase: function () {
```

Comment:
It could easily be simplified to simply test the call of the function. I'll change immediately.